### PR TITLE
Extended the MultipartForm type to allow accessing file uploads

### LINKED ISSUE #32: Implement accessor methods for getting a file passed in a multipart request
A POST, PATCH or PUT request with the header `Content-Type:multipart/form-data` allows users to send files as part of the request. This is currently not implemented in  the `safehttp.MultipartForm` type. We need to investigate the behaviour of the current API, provided by `net/http`, and provide accessor methods for form files in the `safehttp` API

### PR DESCRIPTION
**Pull request by mihalimara22**
Fixes #32, #38

Extended the `MultipartForm` to support retrieving file uploads from incoming requests. This will prevent the user from directly accessing the multipart form but will still allow access to the accessor methods in `multipart` used for opening files.

Fixed the issue that was causing a multipart form to be capped at 32 MB (my misunderstanding of documentation).

This branch  is rebased on top of `grenfeldt-splitting-form-tests` branch  (#125 ).

### CODE REVIEW

**Comment by mattiasgrenfeldt** [comment]
@googlebot I consent.

File: safehttp/form.go
Comment (mattiasgrenfeldt) [inline safehttp/form.go:208]: nit `FileHeader.Open` instead of just `Open`

File: safehttp/form_test.go
Comment (mattiasgrenfeldt) [inline safehttp/form_test.go:530]: Split the two } and put the second one on the next line

File: safehttp/incoming_request.go
Comment (mattiasgrenfeldt) [inline safehttp/incoming_request.go:92]: Newline before `//`

**Review by mattiasgrenfeldt — APPROVED** [review]
Looks good, just a few nits :)

**Review by bsiegert — APPROVED** [review]

File: safehttp/form.go
Comment (kele) [inline safehttp/form.go:210]: nit: why not `File` like the original API (not a slice, but just keeping the name).

File: safehttp/incoming_request_test.go
Comment (kele) [inline safehttp/incoming_request_test.go:222]: got err

Comment (kele) [reply safehttp/incoming_request_test.go:222]: here and other places

Comment (kele) [reply safehttp/incoming_request_test.go:222]: nvm

**Comment by kele** [comment]
One nit, otherwise LGTM. Sync and we're ready to merge.

**Comment by mihalimara22** [comment]
We can't merge until #125 is merged.

**Review by empijei — APPROVED** [review]

**Comment by mihalimara22** [comment]
Rebased on top of master, ready for merge.

**Pull request merged**
